core/src/main/java/org/elasticsearch/action/admin/cluster/snapshots/get/TransportGetSnapshotsAction.java

```diff
@@ -80,23 +80,22 @@ protected void masterOperation(final GetSnapshotsRequest request, ClusterState s
         try {
             final String repository = request.repository();
             List<SnapshotInfo> snapshotInfoBuilder = new ArrayList<>();
+            final Map<String, SnapshotId> allSnapshotIds = new HashMap<>();
+            final List<SnapshotId> currentSnapshotIds = new ArrayList<>();
+            for (SnapshotInfo snapshotInfo : snapshotsService.currentSnapshots(repository)) {
+                SnapshotId snapshotId = snapshotInfo.snapshotId();
+                allSnapshotIds.put(snapshotId.getName(), snapshotId);
+                currentSnapshotIds.add(snapshotId);
+            }
+            for (SnapshotId snapshotId : snapshotsService.snapshotIds(repository)) {
```

Comment:
Thinking about this some more, I like the idea of being able to specify `_current` in addition to other snapshots.  I can imagine a user wanting to know the state of snapshots starting with "2016-11-*" and the "current" one.  In this case, we would treat `_current` as a regex that resolves to whatever the currently running snapshots are.  What do you think about that?